### PR TITLE
Development of the new menu (at the top of the page)

### PR DESCRIPTION
**Pull request by dlumbrer**
This PR adds all the functionality of the new menu. Each commit has its feature/fix isolated and this branch (`menu-top-6.1.0`) will be the branch where the menu is going to be mantained.

### CODE REVIEW

File: src/ui/public/kbn_top_nav/kbn_top_nav.js [hunk 1]
```diff
@@ -35,18 +35,21 @@
 
 import _ from 'lodash';
 import angular from 'angular';
+import { customMenu } from 'ui/kibiter/menu/custom_menu'
```
Comment (sduenas) [inline src/ui/public/kbn_top_nav/kbn_top_nav.js]: Organize these imports along with the style one in a single block to make clear they only belong to kibiter menu functionality.

File: src/ui/public/kbn_top_nav/less/custom_style.less [hunk 2]
```diff
@@ -0,0 +1,46 @@
+.newLocalNav {
```
Comment (sduenas) [inline src/ui/public/kbn_top_nav/less/custom_style.less]: Rename these classes using `kibiter` as prefix.

File: src/ui/public/kibiter/menu/custom_menu.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            countItems++
+        })
+    }
+
+    const closeSubmenu = ($scope) => {
+        $scope.showNewMenu = false;
+        $scope.parentDashboard = undefined;
+    }
+
+    $scope.redirectToPanel = (panel_id) => {
```
Comment (sduenas) [inline src/ui/public/kibiter/menu/custom_menu.js]: Move this function at the same level of the above ones (those that aren't `const` functions)

File: src/ui/public/kibiter/menu/custom_menu.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        item.dashboards.forEach(function (subitem) {
+            if (!$scope.currentPanelsons[countItems]) {
+                $scope.currentPanelsons[countItems] = []
+            }
+            $scope.currentPanelsons[countItems].push(subitem)
+            if (countItems >= numberColumns - 1) {
+                countItems = 0;
+                return
+            }
+            countItems++
```
Comment (sduenas) [inline src/ui/public/kibiter/menu/custom_menu.js]: Put this into an `else` block too.

File: src/ui/public/kibiter/menu/custom_menu.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+
+export function customMenu($scope) {
+    // Define the number of columns
+    let numberColumns = 4
+
+    $scope.showInfo = function (item) {
```
Comment (sduenas) [inline src/ui/public/kibiter/menu/custom_menu.js]: This is the old style javascript for defining functions. Change it to the new style.

File: src/ui/public/kibiter/menu/custom_menu.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        if (item.type === "entry") {
+            $scope.redirectToPanel(item.panel_id)
+        } else if (item.type === "menu") {
+            // If clicked in the same item
+            if ($scope.parentDashboard === item) {
+                closeSubmenu($scope)
+                return
+            }
+            showSubmenu($scope, item)
```
Comment (sduenas) [inline src/ui/public/kibiter/menu/custom_menu.js]: Put this into an else block.

File: src/ui/public/kibiter/menu/get_metadashboard.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
+
+export function getMetadashboard(es, kbnIndex, scope) {
```
Comment (sduenas) [inline src/ui/public/kibiter/menu/get_metadashboard.js]: This function mixes functions of different levels of abstraction. I suggest to:

* Rename this function to `retrieveKibiterMenuData` or something similar.
* Move the code that searches in ElasticSearch to two different functions: `fetchMetadashboar` and `fetchProjectName`.
* Rename `styleSelected` function to `highlightMenu`.
* Call these functions in the body of this one.

File: src/ui/public/kibiter/menu/custom_menu.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
+
+export function customMenu($scope) {
```
Comment (sduenas) [inline src/ui/public/kibiter/menu/custom_menu.js]: Rename this function to `drawKibiterMenu`, `displayKibiterMenu`, `renderKibiterMenu` or something similar.

**Review by sduenas — CHANGES_REQUESTED** [review]
Please address my comments.

**Comment by dlumbrer** [comment]
Done @sduenas , please, check my changes!

Thanks!

**Review by sduenas — APPROVED** [review]
Great. This looks better. Would you squash the commits before merging it?

**Pull request merged**